packages/broker/src/plugins/legacyWebsocket/WebsocketServer.ts

```diff
@@ -16,21 +13,43 @@ import { StreamStateManager } from '../../StreamStateManager'
 import { StorageNodeRegistry } from '../../StorageNodeRegistry'
 import { Stream } from '../../Stream'
 import { StreamFetcher } from '../../StreamFetcher'
+import http from "http"
+import https from "https"
+import { parse as parseQuery } from 'querystring'
 
 const logger = new Logger(module)
 
 export class WebsocketServer extends EventEmitter {
 
-    wss: TemplatedApp
-    _listenSocket: uWS.us_listen_socket|null
-    requestHandler: RequestHandler
-    connections: Map<string,Connection>
-    pingInterval: number
-    metrics: Metrics
-    _pingInterval: NodeJS.Timeout
+    static validateProtocolVersions(controlLayerVersion: number|undefined, messageLayerVersion: number|undefined): void | never {
+        if (controlLayerVersion === undefined || messageLayerVersion === undefined) {
+            throw new Error('Missing version negotiation! Must give controlLayerVersion and messageLayerVersion as query parameters!')
+        }
+
+        // Validate that the requested versions are supported
+        if (ControlLayer.ControlMessage.getSupportedVersions().indexOf(controlLayerVersion) < 0) {
+            throw new Errors.UnsupportedVersionError(controlLayerVersion, `Supported ControlLayer versions: ${
+                JSON.stringify(ControlLayer.ControlMessage.getSupportedVersions())
+            }. Are you using an outdated library?`)
+        }
+
+        if (MessageLayer.StreamMessage.getSupportedVersions().indexOf(messageLayerVersion) < 0) {
+            throw new Errors.UnsupportedVersionError(messageLayerVersion, `Supported MessageLayer versions: ${
+                JSON.stringify(MessageLayer.StreamMessage.getSupportedVersions())
+            }. Are you using an outdated library?`)
+        }
+    }
+
+    private readonly httpServer: http.Server | https.Server
+    private readonly wss: WebSocket.Server
+    private readonly requestHandler: RequestHandler
+    private readonly connections: Set<Connection>
+    private readonly pingIntervalInMs: number
+    private readonly metrics: Metrics
+    private readonly pingInterval: NodeJS.Timeout
 
     constructor(
-        wss: TemplatedApp,
+        httpServer: http.Server | https.Server,
         port: number,
```

Comment:
The `port` parameter is not needed anymore

Reply:
fixed